File: app/src/main/java/io/github/hyuwah/catatanku/splash/SplashActivity.java
package io.github.hyuwah.catatanku.splash;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import io.github.hyuwah.catatanku.R;
import io.github.hyuwah.catatanku.notelist.NoteListActivity;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        final Intent intent = new Intent(this, NoteListActivity.class);

        // Delayed 2s to Main Activity
        Thread launch = new Thread(){
            @Override
            public void run() {
                try{
                    sleep(2000);
                }catch (InterruptedException e){
                    e.printStackTrace();
                }finally {
                    startActivity(intent);
                    finish();
                }
            }
        };

        launch.start();
    }
}

File: app/src/main/java/io/github/hyuwah/catatanku/ui/splash/SplashActivity.kt
package io.github.hyuwah.catatanku.ui.splash

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import io.github.hyuwah.catatanku.databinding.ActivitySplashBinding
import io.github.hyuwah.catatanku.ui.notelist.NoteListActivity
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class SplashActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySplashBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySplashBinding.inflate(layoutInflater)
        setContentView(binding.root)
        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.CREATED) {
                delay(2000)
                startActivity(Intent(this@SplashActivity, NoteListActivity::class.java))
                finish()
            }
        }
    }
}